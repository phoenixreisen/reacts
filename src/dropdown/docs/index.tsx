import { BrowserRouter as Router, Routes, Route } from 'react-router-dom';
import { createRoot } from 'react-dom/client';
import { Dropdown } from '../dropdown.r';
import React from 'react';

export const Root = () => {
    return (
        <Dropdown title="Awesome Dropdown">
            <a href="#">Link 1</a>
            <a href="#">Link 2</a>
            <a href="#">Link 3</a>
        </Dropdown>
    );
};

createRoot(document.querySelector('.example-app')).render(
    <Router>
        <Routes>
            <Route path="*" element={<Root />} />
        </Routes>
    </Router>
);
