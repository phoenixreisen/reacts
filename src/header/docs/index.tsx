import { BrowserRouter as Router, Routes, Route } from 'react-router-dom';
import { createRoot } from 'react-dom/client';
import React, { useState } from 'react';
import { Header, Topbar } from '..';

export const Root = () => {
    const [open, setOpen] = useState<boolean>(false);

    return (<>
        <div className="wrapper wrapper--large">
            <Header
                version="Header mit Topbar Demo"
                toggleNav={(e) => { e.preventDefault(); setOpen(!open) }}
            />
        </div>
        <Topbar
            name="Phoenix Demo App"
            backUrl="https://www.phoenixreisen.com"
            toggleNav={(e) => { e.preventDefault(); setOpen(!open) }}
            toggleAvatar={(e) => { e.preventDefault(); setOpen(!open) }}
        />
    </>);
};

createRoot(document.querySelector('.example-app')).render(
    <Router>
        <Routes>
            <Route path="*" element={<Root />} />
        </Routes>
    </Router>
);
