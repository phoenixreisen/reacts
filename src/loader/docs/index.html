<!DOCTYPE html>
<html lang="de">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <!-- <link rel="stylesheet" type="text/css" href="http://localhost:3027/phx-design-system.css" /> -->
    <link rel="stylesheet" type="text/css" href="https://design-system.phoenixreisen.net/phx-design-system.css" />

    <title>Phoenix Loader Example</title>
</head>
<body>
    <header class="wrapper">
        <div class="example-app"></div>
    </header>

    <script>
        window.process = {
            env: {
                NODE_ENV: 'development'
            }
        }
    </script>
    <script src="./demo.min.js"></script>
</body>
</html>