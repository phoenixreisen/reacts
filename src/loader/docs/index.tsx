import { BrowserRouter as Router, Routes, Route } from 'react-router-dom';
import { createRoot } from 'react-dom/client';
import React from 'react';
import Loader from '..';

export const Root = () => <Loader text="Demo wird geladen..." />;

createRoot(document.querySelector('.example-app')).render(
    <Router>
        <Routes>
            <Route path="*" element={<Root />} />
        </Routes>
    </Router>
);
