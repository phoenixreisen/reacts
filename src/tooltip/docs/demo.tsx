import React from 'react';
import Tooltip from '..';

//--- React Demo Component -----

export const Content = () => {
    return (
        <article className="pa3 gray6 c-gray1" >
            <div className="f6 ma0 mb2">
                <i className="fas fa-home mr2"></i>
                Tooltip mit Ãœberschrift
            </div>
            <p>
                Hier etwas mehr Text. Hier etwas mehr Text. Hier etwas mehr Text.<br />
                Hier etwas mehr Text. Hier etwas mehr Text. Hier etwas mehr Text. 
            </p>
            <div>
                <a href="https://www.phoenixreisen.com">
                    Zu Phoenix Reisen
                </a>
            </div>
        </article>
    );
};

//--- Tooltip Component in several variations -----

export const Demo = () => {
    return (
        <div className="pa4 tc">
            <div className="ma3">
                <Tooltip 
                    event="click"
                    color="warning"
                    position="right"
                    iconname='fa-info-circle'
                    TipComponent={<Content />}
                />
            </div>
            <div className="ma3">
                <Tooltip 
                    color="success"
                    position="left"
                    text="Hover mich!"
                    tooltip="Ich bin der 1. Tooltip!"
                />
             </div>
            <div className="ma3">
                <Tooltip 
                    color="danger"
                    position="right"
                    text="Hover mich auch!"
                    tooltip="Ich bin der 2. Tooltip!"
                />
            </div>
            <div className="ma3">
                <Tooltip 
                    color="info"
                    position="below"
                    text="Mich auch hovern!"
                    tooltip="Ich bin der 3. Tooltip!"
                />
            </div>
            <div className="ma3">
                <Tooltip 
                    color="warning"
                    position="above"
                    text="Hover mich endlich auch!"
                    tooltip="Ich bin der 4. Tooltip!"
                    iconname='fa-exclamation-triangle'
                />
            </div>
            <div className="ma3">
                <Tooltip 
                    event="click"
                    color="warning"
                    position="below"
                    text="Ich rendere eine Komponente"
                    TipComponent={<Content />}
                />
            </div>
        </div>
    );
};