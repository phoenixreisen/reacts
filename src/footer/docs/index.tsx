import { BrowserRouter as Router, Route, Switch } from 'react-router-dom';
import { Demo } from './footer.demo';
import ReactDOM from 'react-dom';
import React from 'react';

export const Root = () => {
    return (
        <Demo />
    );
};

ReactDOM.render(
    <Router>
        <Switch>
            <Route path="/" render={() => <Root />} />
        </Switch>
    </Router>,
    document.querySelector('.example-app'),
);
