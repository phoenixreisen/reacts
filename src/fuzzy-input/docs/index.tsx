import { BrowserRouter as Router, Routes, Route } from 'react-router-dom';
import { createRoot } from 'react-dom/client';
import { Demo } from './fuzzy.demo';
import React from 'react';

export const Root = () => {
    return (
        <Demo />
    );
};

createRoot(document.querySelector('.example-app')).render(
    <Router>
        <Routes>
            <Route path="*" element={<Root />} />
        </Routes>
    </Router>
);
