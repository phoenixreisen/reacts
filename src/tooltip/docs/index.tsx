import { BrowserRouter as Router, Route, Switch } from 'react-router-dom';
import ReactDOM from 'react-dom';
import { Demo } from './demo';
import React from 'react';

ReactDOM.render(
    <Router>
        <Switch>
            <Route path="/" render={() => <Demo />} />
        </Switch>
    </Router>,
    document.querySelector('.example-app'),
);