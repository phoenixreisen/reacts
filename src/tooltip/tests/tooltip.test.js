/**
 * @jest-environment jsdom
 */
import { render, fireEvent, waitFor, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { Content } from '../docs/demo'; 
import Tooltip from '../tooltip';
import React from "react";

describe('Tooltip should', () => {
    test('have some default classes', () => {
        const component = renderer.create(
            <Tooltip text='Tooltip' tooltip='Hinweis!' />
        );
        const snap = component.toJSON();
        expect(snap).toMatchSnapshot();
    });

    test('render text with icon', () => {
        const component = renderer.create(
            <Tooltip text="How to live?!" iconname="fas fa-blabla" tooltip="WORK HARD, PLAY HARD" />
        );
        const snap = component.toJSON();
        expect(snap).toMatchSnapshot();
    });

    test('also render React Components as tooltip', () => {
        const component = renderer.create(
            <Tooltip text="Ich bin ein Tooltip" TipComponent={<Content />} />
        );
        const snap = component.toJSON();
        expect(snap).toMatchSnapshot();
    });

    test('render positions', () => {
        const component = renderer.create(
            <React.Fragment>
                <Tooltip text='Tooltip' tooltip='Hinweis left!' position='left' />
                <Tooltip text='Tooltip' tooltip='Hinweis rechts!' position='right' />
                <Tooltip text='Tooltip' tooltip='Hinweis drunter!' position='below' />
                <Tooltip text='Tooltip' tooltip='Hinweis drüber!' position='above' />
            </React.Fragment>
        );
        const snap = component.toJSON();
        expect(snap).toMatchSnapshot();
    });

    test('render colors', () => {
        const component = renderer.create(
            <React.Fragment>
                <Tooltip text='Tooltip' tooltip='Hinweis grüm!' color='danger' />
                <Tooltip text='Tooltip' tooltip='Hinweis grüm!' color='success' />
                <Tooltip text='Tooltip' tooltip='Hinweis grüm!' color='warning' />
                <Tooltip text='Tooltip' tooltip='Hinweis grüm!' color='info' />
            </React.Fragment>
        );
        const snap = component.toJSON();
        expect(snap).toMatchSnapshot();
    });

    test('also handle on click instead of hover', () => {
        const component = renderer.create(
            <React.Fragment>
                <Tooltip text='Tooltip' tooltip='Hinweis grüm!' event='click' />
                <Tooltip text='Tooltip' TipComponent={<Content />} event='click' />
            </React.Fragment>
        );
        const snap = component.toJSON();
        expect(snap).toMatchSnapshot();
    });

    test('change visibility class on mouse over/leave', () => {
        const component = render(
            <Tooltip key="1" text='Tooltip' tooltip='Hinweis!' />
        );

        expect(document.querySelector('.tip--visible') === null).toBeTruthy();
        fireEvent.mouseEnter(document.querySelector('.tooltip'));
        expect(document.querySelector('.tip--visible') !== null).toBeTruthy();
        fireEvent.mouseLeave(document.querySelector('.tooltip'));
        expect(document.querySelector('.tip--visible') === null).toBeTruthy();
    });

    test('change visibility class on click', () => {
        const component = render(
            <Tooltip key="2" text='Tooltip' tooltip='Hinweis!' event="click" />
        );

        expect(document.querySelector('.tip--visible') === null).toBeTruthy();
        fireEvent.click(document.querySelector('.tooltip'));
        expect(document.querySelector('.tip--visible') !== null).toBeTruthy();
        fireEvent.click(document.querySelector('.tooltip'));
        expect(document.querySelector('.tip--visible') === null).toBeTruthy();
    });
});