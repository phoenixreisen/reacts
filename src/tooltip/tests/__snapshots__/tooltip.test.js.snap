// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tooltip should also handle on click instead of hover 1`] = `
[
  <article
    className="tooltip tooltip--click"
    onClick={[Function]}
  >
    <span
      className="tooltip-text"
    >
      Tooltip
    </span>
    <span
      className="tip  tip--below tip--hidden"
    >
      <span
        className=""
      >
        Hinweis grüm!
      </span>
    </span>
  </article>,
  <article
    className="tooltip tooltip--click"
    onClick={[Function]}
  >
    <span
      className="tooltip-text"
    >
      Tooltip
    </span>
    <span
      className="tip tip--component tip--below tip--hidden"
    >
      <article
        className="pa3 gray6 c-gray1"
      >
        <div
          className="f6 ma0 mb2"
        >
          <i
            className="fas fa-home mr2"
          />
          Tooltip mit Überschrift
        </div>
        <p>
          Hier etwas mehr Text. Hier etwas mehr Text. Hier etwas mehr Text.
          <br />
          Hier etwas mehr Text. Hier etwas mehr Text. Hier etwas mehr Text.
        </p>
        <div>
          <a
            href="https://www.phoenixreisen.com"
          >
            Zu Phoenix Reisen
          </a>
        </div>
      </article>
    </span>
  </article>,
]
`;

exports[`Tooltip should also render React Components as tooltip 1`] = `
<article
  className="tooltip "
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <span
    className="tooltip-text"
  >
    Ich bin ein Tooltip
  </span>
  <span
    className="tip tip--component tip--below tip--hidden"
  >
    <article
      className="pa3 gray6 c-gray1"
    >
      <div
        className="f6 ma0 mb2"
      >
        <i
          className="fas fa-home mr2"
        />
        Tooltip mit Überschrift
      </div>
      <p>
        Hier etwas mehr Text. Hier etwas mehr Text. Hier etwas mehr Text.
        <br />
        Hier etwas mehr Text. Hier etwas mehr Text. Hier etwas mehr Text.
      </p>
      <div>
        <a
          href="https://www.phoenixreisen.com"
        >
          Zu Phoenix Reisen
        </a>
      </div>
    </article>
  </span>
</article>
`;

exports[`Tooltip should have some default classes 1`] = `
<article
  className="tooltip "
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <span
    className="tooltip-text"
  >
    Tooltip
  </span>
  <span
    className="tip  tip--below tip--hidden"
  >
    <span
      className=""
    >
      Hinweis!
    </span>
  </span>
</article>
`;

exports[`Tooltip should render colors 1`] = `
[
  <article
    className="tooltip "
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <span
      className="tooltip-text"
    >
      Tooltip
    </span>
    <span
      className="tip  tip--below tip--hidden"
    >
      <span
        className="tip--colored danger"
      >
        Hinweis grüm!
      </span>
    </span>
  </article>,
  <article
    className="tooltip "
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <span
      className="tooltip-text"
    >
      Tooltip
    </span>
    <span
      className="tip  tip--below tip--hidden"
    >
      <span
        className="tip--colored success"
      >
        Hinweis grüm!
      </span>
    </span>
  </article>,
  <article
    className="tooltip "
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <span
      className="tooltip-text"
    >
      Tooltip
    </span>
    <span
      className="tip  tip--below tip--hidden"
    >
      <span
        className="tip--colored warning"
      >
        Hinweis grüm!
      </span>
    </span>
  </article>,
  <article
    className="tooltip "
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <span
      className="tooltip-text"
    >
      Tooltip
    </span>
    <span
      className="tip  tip--below tip--hidden"
    >
      <span
        className="tip--colored info"
      >
        Hinweis grüm!
      </span>
    </span>
  </article>,
]
`;

exports[`Tooltip should render positions 1`] = `
[
  <article
    className="tooltip "
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <span
      className="tooltip-text"
    >
      Tooltip
    </span>
    <span
      className="tip  tip--left tip--hidden"
    >
      <span
        className=""
      >
        Hinweis left!
      </span>
    </span>
  </article>,
  <article
    className="tooltip "
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <span
      className="tooltip-text"
    >
      Tooltip
    </span>
    <span
      className="tip  tip--right tip--hidden"
    >
      <span
        className=""
      >
        Hinweis rechts!
      </span>
    </span>
  </article>,
  <article
    className="tooltip "
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <span
      className="tooltip-text"
    >
      Tooltip
    </span>
    <span
      className="tip  tip--below tip--hidden"
    >
      <span
        className=""
      >
        Hinweis drunter!
      </span>
    </span>
  </article>,
  <article
    className="tooltip "
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <span
      className="tooltip-text"
    >
      Tooltip
    </span>
    <span
      className="tip  tip--above tip--hidden"
    >
      <span
        className=""
      >
        Hinweis drüber!
      </span>
    </span>
  </article>,
]
`;

exports[`Tooltip should render text with icon 1`] = `
<article
  className="tooltip "
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <i
    className="tooltip-icon fas fas fa-blabla mr2"
  />
  <span
    className="tooltip-text"
  >
    How to live?!
  </span>
  <span
    className="tip  tip--below tip--hidden"
  >
    <span
      className=""
    >
      WORK HARD, PLAY HARD
    </span>
  </span>
</article>
`;
